tests/nativeapp/codegen/templating/test_templating_processor.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from unittest import mock
+
+import pytest
+from snowflake.cli._plugins.nativeapp.bundle_context import BundleContext
+from snowflake.cli._plugins.nativeapp.codegen.templating.templating_processor import (
+    TemplatingProcessor,
+)
+from snowflake.cli._plugins.nativeapp.exceptions import InvalidTemplateInFileError
+from snowflake.cli.api.exceptions import InvalidTemplate
+from snowflake.cli.api.project.schemas.native_app.path_mapping import PathMapping
+
+from tests.nativeapp.utils import CLI_GLOBAL_TEMPLATE_CONTEXT
+
+
+class FilesSetup(TemporaryDirectory):
```

Comment:
why does it inherit from temp dir? Seems like unnecessary coupling at first glance. I'm not even sure why this needs to be a context manager. Can you explain the thought process?

Reply:
refactored.